### PR TITLE
Add elastic ssl support

### LINKED ISSUE #2047: Elasticsearch at custom location
**Description of problem:**

We would like to run elasticsearch on a custom location at a subdirectory with custom certificates (e.g. https://example.com/elasticsearch/). psort.py does not yet support this. A possible solution is shown below, but it will break stuff. 

**Command line and arguments:**
```sh
psort.py --status_view linear -o elastic --index_name test --doc_type plaso --server example.com:443/elasticsearch/ test.plaso
```
**Solution draft:**
```diff
diff --git a/plaso/output/shared_elastic.py b/plaso/output/shared_elastic.py
index f6ca6178..668c67ca 100644
--- a/plaso/output/shared_elastic.py
+++ b/plaso/output/shared_elastic.py
@@ -60,14 +60,14 @@ class SharedElasticsearchOutputModule(interface.OutputModule):

   def _Connect(self):
     """Connects to an Elasticsearch server."""
-    elastic_hosts = [{'host': self._host, 'port': self._port}]
+    elastic_hosts = [self._host]

     elastic_http_auth = None
     if self._username is not None:
       elastic_http_auth = (self._username, self._password)

     self._client = elasticsearch.Elasticsearch(
-        elastic_hosts, http_auth=elastic_http_auth)
+        elastic_hosts, http_auth=elastic_http_auth, use_ssl=True, verify_certs=True, ca_certs="/etc/ssl/certs/ca-certificates.crt")

     logger.debug('Connected to Elasticsearch server: {0:s} port: {1:d}.'.format(
         self._host, self._port))
```

**Plaso version:**
20180630

### PR DESCRIPTION
**Pull request by cugu**
## One line description of pull request

Add ssl support (with custom certificates) for elastic search

## Description:

- Remove default elasticsearch  port from plaso, it is still the 9200 by default from the elasticsearch python package. This way hosts can be defined by urls like `https://example.com:1234/var`
- Add ca_cert command line option for elasticsearch. This enables usage of custom certificates.

**Related issue (if applicable):** fixes #2047

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/l2tdocs/blob/master/process/Code%20review%20process.asciidoc). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://github.com/log2timeline/plaso/wiki/Style-guide).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
  - [x] Automated checks (Travis, Codecov, Codefactor )pass
  - [x] No new [new dependencies](https://github.com/log2timeline/plaso/wiki/Adding-a-new-dependency) are required or l2tdevtools has been updated
  - [x] Reviewer assigned

### CODE REVIEW

**Comment by cugu** [comment]
The _Connect method for elasticsearch is extended, but not tested, so coverage decreases.

**Comment by joachimmetz** [comment]
Thx for the PR, I've assigned @Onager as the reviewer. Know that both of us have limited availability at the moment but we'll try to review as soon as time permits.

File: plaso/output/shared_elastic.py [hunk 1]
```diff
@@ -65,20 +65,27 @@ def __init__(self, output_mediator):
...
 
     elastic_http_auth = None
     if self._username is not None:
       elastic_http_auth = (self._username, self._password)
 
     self._client = elasticsearch.Elasticsearch(
-        elastic_hosts, http_auth=elastic_http_auth)
+        elastic_hosts,
+        http_auth=elastic_http_auth,
+        ca_certs=self._ca_certs
```
Comment (Onager) [inline plaso/output/shared_elastic.py]: We'd also need to pass verify_certs=True, as I read the API: https://elasticsearch-py.readthedocs.io/en/master/connection.html?highlight=Connection

File: plaso/cli/helpers/elastic_output.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ class ElasticSearchServerArgumentsHelper(server_config.ServerArgumentsHelper):
   """Elastic Search server CLI arguments helper."""
 
   _DEFAULT_SERVER = '127.0.0.1'
-  _DEFAULT_PORT = 9200
+  _DEFAULT_PORT = None
```
Comment (Onager) [inline plaso/cli/helpers/elastic_output.py]: Please remove the port changes from this PR, to keep it focused on just the SSL certificate changes.

File: plaso/output/shared_elastic.py (hunk 1 above)
Comment (cugu) [reply plaso/output/shared_elastic.py]: verify_certs is True by default: https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/connection/http_requests.py#L30

File: plaso/cli/helpers/elastic_output.py (hunk 2 above)
Comment (cugu) [reply plaso/cli/helpers/elastic_output.py]: If the port is set here the `_Connect` functions in `shared_elastic.py` will always use `elastic_hosts = [{'host': self._host, 'port': self._port}]`. This way ssl, subdirectories and other ports cannot be set. The default port in elasticsearch-py is 9200, so I does not need to be set here.

**Comment by cugu** [comment]
Merged the updates and added a use_ssl flag.

**Comment by Onager** [comment]
Made a few minor changes, will merge once tests have passed

**Comment by cugu** [comment]
Very cool, thanks :)

**Pull request merged**